plugins/CustomDimensions/Dimension/Extraction.php

```diff
@@ -128,7 +128,7 @@ private function formatPattern()
 
         $pattern = $this->pattern;
         if ($this->dimension === 'urlparam') {
-            $pattern = '\?.*' . $pattern . '=([^&]*)';
```

Comment:
The change actually removes the requirement to have the param in the query string, as a `?` is no longer be required. It would e.g. also match something like `http://example.com/myfile&param=x/start`
So maybe something like this would be more accurate: `\?(?:.*&)?`

Reply:
sure, it's an invalid URL if the `&` isn't escaped like that but I can change it to `\?(?:.*&)?` to be more strict